.github/workflows/run-unit-tests.yml

```diff
@@ -1,32 +1,52 @@
-# WARNING! It is currently being investigated how to make this faster
-# DO NOT blindly copy this workflow, not noticing the slow down,
-# because suddenly our tests will take hours to pass CI.
-# Hopefully this comment here will help prevent that.
-# https://github.com/MetaMask/metamask-extension/issues/25680
-
 name: Run unit tests
 
 on:
+  push:
+    branches: [develop, master]
   pull_request:
     types: [opened,reopened,synchronize]
 
 jobs:
-  test-unit-jest:
+  test-unit:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        shard: [1, 2, 3, 4, 5, 6, 7, 8]
```

Comment:
I would slightly reduce sharding.  With one shard, the compute time is approximately 28 minutes.  With 8 shards, the compute time is approximately 48 minutes.  Time spent in `Setup environment` is fairly high, so more sharding has diminishing returns.  Maybe try 6 shards?

Reply:
6 shards: total workflow duration: 7m 12s, total compute time: 35m
5 shards: total workflow duration: 7m 48s, total compute time: 34m

Reply:
@HowardBraham  I think we can stick to 6 shards, let me know what you think.

Reply:
:+1: